firefox-ios/Client/Frontend/PasswordGenerator/PasswordGeneratorViewController.swift

```diff
@@ -0,0 +1,293 @@
+                constant: -UX.containerPadding),
+            contentView.topAnchor.constraint(
+                equalTo: view.safeAreaLayoutGuide.topAnchor,
+                constant: UX.containerPadding),
+            contentView.bottomAnchor.constraint(
+                equalTo: view.safeAreaLayoutGuide.bottomAnchor,
+                constant: -UX.containerPadding),
+        ])
+
+        // Content View Elements Layout
+        NSLayoutConstraint.activate([
+            header.leadingAnchor.constraint(equalTo: contentView.leadingAnchor),
+            header.trailingAnchor.constraint(
+                equalTo: contentView.trailingAnchor,
+                constant: -UX.headerTrailingPadding),
+            header.topAnchor.constraint(equalTo: contentView.topAnchor),
+            descriptionLabel.leadingAnchor.constraint(equalTo: contentView.leadingAnchor),
+            descriptionLabel.trailingAnchor.constraint(equalTo: contentView.trailingAnchor),
+            descriptionLabel.topAnchor.constraint(
+                equalTo: header.bottomAnchor,
+                constant: UX.containerElementsVerticalPadding),
+            passwordField.leadingAnchor.constraint(equalTo: contentView.leadingAnchor),
+            passwordField.trailingAnchor.constraint(equalTo: contentView.trailingAnchor),
+            passwordField.topAnchor.constraint(
+                equalTo: descriptionLabel.bottomAnchor,
+                constant: UX.containerElementsVerticalPadding),
+            usePasswordButton.leadingAnchor.constraint(equalTo: contentView.leadingAnchor),
+            usePasswordButton.trailingAnchor.constraint(
+                equalTo: passwordField.trailingAnchor),
+            usePasswordButton.topAnchor.constraint(
+                equalTo: passwordField.bottomAnchor,
+                constant: UX.containerElementsVerticalPadding),
+            usePasswordButton.bottomAnchor.constraint(equalTo: contentView.bottomAnchor)
+        ])
+
+        // Header elements layout
+        NSLayoutConstraint.activate([
+            headerImageView.leadingAnchor.constraint(equalTo: header.leadingAnchor),
+            headerImageView.centerYAnchor.constraint(equalTo: headerLabel.centerYAnchor),
+            headerImageView.widthAnchor.constraint(equalToConstant: scaledHeaderImageSize),
+            headerImageView.heightAnchor.constraint(equalToConstant: scaledHeaderImageSize),
+            headerLabel.leadingAnchor.constraint(
+                equalTo: headerImageView.trailingAnchor,
+                constant: UX.headerIconLabelSpacing),
+            headerLabel.trailingAnchor.constraint(equalTo: header.trailingAnchor),
+            headerLabel.topAnchor.constraint(
+                equalTo: header.topAnchor,
+                constant: UX.headerVerticalPadding),
+            headerLabel.bottomAnchor.constraint(equalTo: header.bottomAnchor)
+        ])
+
+        // Password Field Elements Layout
+        NSLayoutConstraint.activate([
```

Comment:
Since a few of these views seem self-contained (i.e. passwordField), to avoid the VC from being large, we may be able to break these views into its own classes and files + add the constraints there. This is just a preference, but not necessarily a blocker on the PR. Let me know what you think!

Reply:
Great idea, I think extracting the header and passwordField views into their own classes would be good

Reply:
awesome thanks!